src/disk/config/disks.rs

```diff
@@ -251,7 +251,7 @@ impl Disks {
     pub fn unmount_devices(&self) -> Result<(), DiskError> {
         let mounts = Mounts::new().unwrap();
         for device in self.get_physical_devices() {
-            for mount in mounts.starts_with(device.get_device_path().as_os_str().as_bytes()) {
+            for mount in mounts.exactly(device.get_device_path().as_os_str().as_bytes()) {
```

Comment:
This is perhaps a dangerous change

Reply:
Only one way to find out. I'll start testing this.

Reply:
Is this even needed? Seems that simply checking if the device has the mount point field set would be enough.